### PR TITLE
bpf: Quieten mock targets

### PR DESCRIPTION
**Pull request by joestringer**
Make the bpf mock testing framework targets respect the user's verbosity
flag.

### CODE REVIEW

File: test/bpf/Makefile [hunk 1]
```diff
@@ -43,16 +43,16 @@ elf-demo.o: elf-demo.c
...
 
 drop-notify-test: drop-notify-test.c $(LIB)
 	@$(ECHO_CC)
-	make -C $(ROOT_DIR)/bpf/mock generate_helper_headers
-	make -C $(ROOT_DIR)/bpf/mock mock_helpers
-	$(QUIET) $(DOCKER_RUN) $(CLANG) $(FLAGS) -I../../bpf/mock -I $../../bpf/ -I../../../CMock/src -I../../../CMock/vendor/unity/src  $< ../../../CMock/vendor/unity/src/unity.c ../../../CMock/src/cmock.c ../../bpf/mock/mocks/mock_helpers.c ../../bpf/mock/mocks/mock_helpers_skb.c -o $@
+	$(QUIET)$(MAKE) -C $(ROOT_DIR)/bpf/mock generate_helper_headers
+	$(QUIET)$(MAKE) -C $(ROOT_DIR)/bpf/mock mock_helpers
+	$(QUIET)$(QUIET) $(DOCKER_RUN) $(CLANG) $(FLAGS) -I../../bpf/mock -I $../../bpf/ -I../../../CMock/src -I../../../CMock/vendor/unity/src  $< ../../../CMock/vendor/unity/src/unity.c ../../../CMock/src/cmock.c ../../bpf/mock/mocks/mock_helpers.c ../../bpf/mock/mocks/mock_helpers_skb.c -o $@
```
Comment (borkmann) [inline test/bpf/Makefile]: `$(QUIET)$(QUIET)` needed twice? Otherwis lgtm.

**Review by borkmann — APPROVED** [review]

**Review by jibi — APPROVED** [review]

File: test/bpf/Makefile (hunk 1 above)
Comment (joestringer) [reply test/bpf/Makefile]: Woops. Fixed.

File: bpf/mock/Makefile [hunk 2]
```diff
@@ -22,13 +22,13 @@ check_helper_headers: generate_helper_headers mock_helpers
...
 	# The temp headers are removed after finishing.
-	rm -r mocks;
+	$(QUIET)rm -r mocks;
 
 generate_helper_headers:
 	# Auto-generate helpers.h, helpers_skb.h, and helpers_xdp.h.
 	# Take contents from bpf/helpers.h, bpf/helpers_skb.h and bpf/helpers_xdp.h.
 	# Prune to be ready to mock and save to the current directory.
-	mkdir -p mocks; \
+	$(QUIET)mkdir -p mocks; \
```
Comment (qmonnet) [inline bpf/mock/Makefile]: Nit: Would be nice to indent the `sed` lines below, I didn't understand at first that their were the continuation of the same command and was wondering why they didn't get their `$(QUIET)`.

**Review by qmonnet — APPROVED** [review]
Somehow I got triggered on the `Quieten` -- but it's not a typo this time, all good :) PR looks good as well.

File: bpf/mock/Makefile (hunk 2 above)
Comment (joestringer) [reply bpf/mock/Makefile]: I'm actually not sure why there's all these `;` throughout this Makefile, probably the original author didn't fully understand the Makefile syntax and how this causes the rest of the commands to be shelled out and to ignore the result of each command. Would be a nice cleanup but yeah seems like we could follow up separately on this.

**Comment by joestringer** [comment]
None of the failures make sense since this is only changing makefile targets for the bpf mock tests.

File: bpf/mock/Makefile (hunk 2 above)
Comment (qmonnet) [reply bpf/mock/Makefile]: Right, didn't pay attention to that but now that you mention it, they should be separate commands entirely.
Agreed that this is for a future PR :+1:.

**Comment by qmonnet** [comment]
/test-1.16-netnext

**Comment by qmonnet** [comment]
/test-1.16-netnext

**Comment by qmonnet** [comment]
/test-1.21-5.4

**Comment by qmonnet** [comment]
/test-1.22-4.19

**Comment by qmonnet** [comment]
/test-1.23-4.9

**Comment by joestringer** [comment]
`test-1.16-net-next` failures look related to provisioning, possibly because I haven't rebased this recently.
`test-1.22-4.19` failure is a known flaky test which has just been quarantined.
Not sure what happened with the AKS run but the sysdump artifacts have expired so I can't dig deeper.

I'll rebase but overall I don't think we need to run the CI on this PR, it's not changing any of the code which is tested by jenkins.

File: bpf/mock/Makefile (hunk 2 above)
Comment (joestringer) [reply bpf/mock/Makefile]: ^^ I've added a new commit on top to resolve this as well, PTAL.

**Comment by joestringer** [comment]
The bpf mock tests should be run as part of the `integration-tests` target, run by Travis. If Travis is happy and there's no further reviews, we should be good to merge this PR.

**Pull request merged**